src/mon/OSDMonitor.cc

```diff
@@ -8685,7 +8685,13 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
      mempool::osdmap::map<string,map<string,string>> old_ec_profiles =
        osdmap.get_erasure_code_profiles();
      auto ec_profiles = pending_inc.get_erasure_code_profiles();
+#ifdef HAVE_STDLIB_MAP_SPLICING
      ec_profiles.merge(old_ec_profiles);
+#else
+     for (auto &i: old_ec_profiles) {
```

Comment:
i'd suggest follow other places where we workaround missing `std::map.merge()` , like:

```c++
ec_profiles.insert(make_move_iterator(begin(old_ec_profiles)),
                   make_move_iterator(end(old_ec_profiles)));
```

Reply:
@wjwithagen ping?